### PR TITLE
fix(pubsub): allow updating topic schema fields individually

### PR DESCRIPTION
**Pull request by hongalex**
Currently, the entirety of `SchemaSettings` must be updated when calling `UpdateTopic`. Instead, allow updating these fields individually (e.g. first/last revision) without needing to specify schema ID or encoding.

### CODE REVIEW

**Review by feywind — APPROVED** [review]

File: pubsub/pstest/fake.go [hunk 1]
```diff
@@ -368,8 +368,26 @@ func (s *GServer) UpdateTopic(_ context.Context, req *pb.UpdateTopicRequest) (*p
...
+			if t.proto.SchemaSettings == nil {
+				t.proto.SchemaSettings = &pb.SchemaSettings{}
+			}
+			t.proto.SchemaSettings.FirstRevisionId = req.Topic.SchemaSettings.FirstRevisionId
+		case "schema_settings.last_revision_id":
+			if t.proto.SchemaSettings == nil {
+				t.proto.SchemaSettings = &pb.SchemaSettings{}
+			}
+			t.proto.SchemaSettings.LastRevisionId = req.Topic.SchemaSettings.LastRevisionId
 		default:
```
Comment (kamalaboulhosn) [inline pubsub/pstest/fake.go:391]: It is also possible to just have schema_settings, which will update all of them. Though I guess the Go library is setting each field explicitly?

File: pubsub/topic.go [hunk 2]
```diff
@@ -407,7 +407,18 @@ func (t *Topic) updateRequest(cfg TopicConfigToUpdate) *pb.UpdateTopicRequest {
 	}
 	if cfg.SchemaSettings != nil {
```
Comment (kamalaboulhosn) [inline pubsub/topic.go:408]: How does one remove the schema config entirely?

**Review by kamalaboulhosn — CHANGES_REQUESTED** [review]

File: pubsub/topic.go (hunk 2 above)
Comment (hongalex) [reply pubsub/topic.go:410]: Yeah I missed this. I added a section that handles clearing SchemaSettings when it detects the zero value, similar to MessageStoragePolicy.

File: pubsub/pstest/fake.go (hunk 1 above)
Comment (hongalex) [reply pubsub/pstest/fake.go:394]: Yeah each field is set explicitly so updating all of them at once is possible too.

**Review by hongalex — COMMENTED** [review]
Addressed comments. Also, kokoro is failing because of a unrelated sample issue, not because of any tests related to pubsub

**Review by kamalaboulhosn — APPROVED** [review]

**Pull request merged**